agents/ansari.py

```diff
@@ -37,10 +36,13 @@ def __init__(self, settings, message_logger=None, json_format=False):
         self.settings = settings
         sq = SearchQuran(settings.KALEMAT_API_KEY.get_secret_value())
         sh = SearchHadith(settings.KALEMAT_API_KEY.get_secret_value())
-        sm = SearchMawsuah(
-            settings.VECTARA_AUTH_TOKEN.get_secret_value(),
-            settings.VECTARA_CUSTOMER_ID,
-            settings.VECTARA_CORPUS_ID,
+        sm = SearchVectara(
```

Comment:
Love this generalization and refactoring.

Reply:
Thanks :)